src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -493,21 +493,27 @@ internal static string TupleMemberName(int position)
             return "Item" + position;
         }
 
-        // This array needs to be sorted for binary searching
-        private static readonly string[] reservedMemberNames = new[]
-        {
-            "CombineHashCodes",
-            "CompareTo",
-            "Create",
-            "Deconstruct",
-            "Equals",
-            "GetHashCode",
-            "GetHashCodeCore",
-            "Rest",
-            "Size",
-            "ToString",
-            "ToStringEnd"
-        };
+        private static bool IsElementNameForbidden(string name)
+        {
+            switch (name)
+            {
+                case "CombineHashCodes":
+                case "CompareTo":
+                case "Create":
+                case "Deconstruct":
+                case "Equals":
+                case "GetHashCode":
+                case "GetHashCodeCore":
+                case "Rest":
+                case "Size":
+                case "ToString":
+                case "ToStringEnd":
```

Comment:
From discussion with Aleksey, "ToStringEnd" can be removed, as it doesn't risk creating ambiguity. I'll remove.